### ISSUE TITLE
In non-existent directory, reloading shell prompt -> error LoadConfig() Getwd failed: "getwd: no such file or directory"

### ISSUE THREAD

**Issue by rlue** [direnv/direnv#455]
## Issue description

When the current working directory does not exist, loading the shell prompt (_e.g.,_ by hitting `<Enter>`) causes the following error to appear:

```
direnv: error LoadConfig() Getwd failed: "getwd: no such file or directory"
```

### Steps to reproduce

```sh
$ mkdir foo
$ cd foo
$ rmdir $(pwd)
direnv: error LoadConfig() Getwd failed: "getwd: no such file or directory"
```

**Comment by zimbatm** [comment]
what is the behaviour that you would expect?

**Comment by rlue** [comment]
Whoops, good point.

As a general principle, I would expect to see an error message only when a program has failed to perform one of its expected functions. Since I don't expect direnv to do anything in a non-existent directory, I would expect to see no error message at all.

**Comment by henrebotha** [comment]
I agree, direnv shouldn't be doing anything (from a user's point of view) in a directory that doesn't exist.

**Comment by HaleTom** [comment]
I'd suggest:

1. Store the CWD when running the hook (initialize to /?)
1. If the stored CWD directory doesn't exist next time, exit early

**Comment by xerus2000** [comment]
> I'd suggest:
> 
>     1. Store the CWD when running the hook (initialize to /?)
> 
>     2. If the stored CWD directory doesn't exist next time, exit early

No need to store anything. Simply catch this error or check for existence beforehand.

**Comment by W1M0R** [comment]
When I experienced this error, it turned out to be related to this WSL issue: https://github.com/microsoft/WSL/issues/6448

This worked for me:

> The workaround is to leave the directory and re-enter it:
> `cd; cd -`
